pkg/danmep/calico.go

```diff
@@ -0,0 +1,36 @@
+package danmep
+
+import (
+    "context"
+    "fmt"
+    "log"
+    "net"
+    "os/exec"
+
+    danmtypes "github.com/nokia/danm/crd/apis/danm/v1"
+    clientset "github.com/nokia/danm/crd/client/clientset/versioned"
+    "github.com/nokia/danm/pkg/ipam"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type calicoDeleteServiceImpl struct {}
+
+func (h *calicoDeleteServiceImpl) WasIPAllocatedByMe(danmClient clientset.Interface, ep *danmtypes.DanmEp, dnet *danmtypes.DanmNet) bool {
+    return ! ipam.WasIpAllocatedByDanm(ep.Spec.Iface.Address, dnet.Spec.Options.Cidr) &&
+        ! ipam.WasIpAllocatedByDanm(ep.Spec.Iface.AddressIPv6, dnet.Spec.Options.Pool6.Cidr) &&
+        ep.Spec.NetworkType == "calico"
+}
+
+func (h *calicoDeleteServiceImpl) DeleteDanmEp(danmClient clientset.Interface, ep *danmtypes.DanmEp, dnet *danmtypes.DanmNet) (err error) {
+    parsedIp := net.ParseIP(ep.Spec.Iface.Address)
```

Comment:
Calico IPAM can handle IPv6 and DS too IMO, so you should prob check both Addess a,d AddressIPv6

Reply:
note: even if you don't want to change the per DanmEp haldning generally speaking (in which case please antoher TODO), inside the Calico handler V6 and dual-stak should be still properly cleant-up though

Reply:
having both V4 and V6 address in a Calico DanmEp totally is still tho IMO

Reply:
I eventually separated IPv4 and IPv6 IP handling